code/ZAS/Zone.dm

```diff
@@ -120,6 +137,9 @@ Class Procs:
 	c_invalidate()
 	for(var/turf/simulated/T in contents)
 		T.update_graphic(graphic_remove = air.graphic) //we need to remove the overlays so they're not doubled when the zone is rebuilt
+		#ifdef ZAS_COLOR
+		T.color = null
+		#endif
 		//T.dbg(invalid_zone)
 		T.needs_air_update = 0 //Reset the marker so that it will be added to the list.
```

Comment:
I have absolutely no idea why this is necessary, but it is. Firelocks get pretty fucky without it.